### ISSUE TITLE
proposal: cascade delete taxonomy -> terms

### ISSUE THREAD

**Issue by MatteoPiovanelli** [OrchardCMS/Orchard#8736]
When deleting a taxonomy, all terms in it should also be deleted: these operations should probably be "gated" by checks to make sure references in the system don't break.

Example of what I have in mind.
![image](https://github.com/OrchardCMS/Orchard/assets/19283945/85a5fff3-d876-45b1-a882-632cf4827f83)
This is a screenshot of a taxonomy in a tenant on my dev box. 
When I hit delete:

- The UI should ask for confirmation
- `OnRemoving<TaxonomyPart>()` it should check whether the taxonomy is selected in the settings of any `TaxonomyField`: if that is the case, the operation should fail, and report the reason (e.g. "Taxonomy is selected in field 'foo' in Content Type 'bar'")
- If the taxonomy may be deleted so far, it should invoke the Remove for all Terms belonging to it.
- 'OnRemoving<TermPart>()' it should check whether the term is selected in any TaxonomyField: if that is the case, the depending on a setting on the ContentType for the term, the operation may fail.

cc @sebastienros  @HermesSbicego-Laser

**Comment by sebastienros** [comment]
Sounds good

**Comment by BenedekFarkas** [comment]
Can this

> 'OnRemoving()' it should check whether the term is selected in any TaxonomyField: if that is the case, the depending on a setting on the ContentType for the term, the operation may fail.

be true if the parent Taxonomy is not selected for a field?

**Comment by MatteoPiovanelli** [comment]
> be true if the parent Taxonomy is not selected for a field?

yes, it may be true even if the taxonomy is not selected for a field. For example, in the case of a tenant with multiple Cultures, in the definition of a `TaxonomyField` you' likely select the taxonomy in you main culture; if later you delete a translation of that taxonomy, you may find its terms have been selected in fields whose "selected" taxonomy is a different one.
I can't guarantee that's the only case, but that's one off the top of my head.

**Comment by BenedekFarkas** [comment]
I see, thanks! I had a hunch it has something to do with translations, especially since you suggested it. :)
